common/src/main/java/org/conscrypt/HpkeAlgorithmIdentifier.java

```diff
@@ -0,0 +1,261 @@
+package org.conscrypt;
+
+import static org.conscrypt.Preconditions.checkNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.crypto.Cipher;
+
+/**
+ * KEM, KDF, and AEAD algorithm identifiers and specs defined on RFC 9180.
+ * <ul>
+ *   <li><a href="https://www.rfc-editor.org/rfc/rfc9180.html#name-key-encapsulation-mechanism">KEM</a></li>
+ *   <li><a href="https://www.rfc-editor.org/rfc/rfc9180.html#name-key-derivation-functions-kd">KDF</a></li>
+ *   <li><a href="https://www.rfc-editor.org/rfc/rfc9180.html#name-authenticated-encryption-wi">AEAD</a></li>
+ * </ul>
+ */
+public class HpkeAlgorithmIdentifier {
+
+    public enum KEM {
+        DHKEM_P_256_HKDF_SHA256, DHKEM_P_384_HKDF_SHA384, DHKEM_P_521_HKDF_SHA512, DHKEM_X25519_HKDF_SHA256, DHKEM_X448_HKDF_SHA512;
```

Comment:
Is there a reason to define the other constants when we don't support them anyway? Ditto for KDF and AEAD. Because these tables are extensible (i.e. more may be defined later), you already cannot assume you've gotten all of them. So it seems better to scope this to just the ones you implement, rather than trying to pull every possible HPKE suite into scope.

Just like how we don't copy over a snapshot of TLS cipher suites into Conscrypt, just the ones we support.